src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/AttachmentController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Controller\Admin\Sell\Catalog\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Attachment\Query\GetProductAttachmentInfo;
+use PrestaShop\PrestaShop\Core\Domain\Product\Attachment\QueryResult\ProductAttachmentInfo;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+/**
+ * Responsible for attachments in Product page
+ */
+class AttachmentController extends FrameworkBundleAdminController
+{
+    /**
+     * @AdminSecurity("is_granted(['read'], 'AdminProducts') || is_granted('create', 'AdminAttachments')")
+     *
+     * @param int $attachmentId
+     *
+     * @return JsonResponse
+     */
+    public function getAttachmentInfoAction(int $attachmentId): JsonResponse
+    {
+        $attachmentInfo = $this->getQueryBus()->handle(new GetProductAttachmentInfo($attachmentId));
+
+        return $this->json(['attachmentInfo' => $this->presentAttachmentInfo($attachmentInfo)]);
```

Comment:
Not sure we need this extra layer `'attachmentInfo'` since it's only an API that returns the attachment object

Reply:
Hmm I don't mind it being here, its like object name and it will be easier to add something if we want in future.